test/algolia/middleware/retry_test.exs

```diff
@@ -13,40 +13,88 @@ defmodule Algolia.Middleware.RetryTest do
 
   test "returns response with successful status code", %{client: client} do
     Tesla.Mock.mock(fn
-      %{url: "/1/indexes", opts: [curr_retry: 0]} ->
+      %{url: "/1/indexes"} ->
+        {200, [], "OK"}
+    end)
+
+    assert {:ok, %{body: "OK"}} = Tesla.get(client, "/1/indexes", opts: [subdomain_hint: :read])
+  end
+
+  test "does not retry 400-level HTTP errors", %{client: client} do
+    Tesla.Mock.mock(fn
+      %{url: "/1/indexes", opts: [retry_count: 3, subdomain_hint: :read]} ->
```

Comment:
If i'm reading this right, this clause's role is to never be called? Perhaps we could just assert on `opts` instead, if `Tesla.Mock` doesn't support single shot mocks.

Reply:
It is. You're right, we could just assert on the `opts` of the response env, making sure it doesn't include a `retry_count`.

And yeah, `Tesla.Mock` is _very_ basic, unlike what we can do with Mox. No single shot expectations or sequencing or anything like that.